graph_weather/models/analysis.py

```diff
@@ -90,6 +114,28 @@ def __init__(
             use_checkpointing=use_checkpointing,
         )
 
+    @classmethod
+    def from_config(cls, config_dict: dict):
+        """Create GraphWeatherAssimilator from configuration dictionary using dacite."""
+        config = dacite.from_dict(data_class=GraphWeatherAssimilatorConfig, data=config_dict)
+        return cls(
+            output_lat_lons=config.output_lat_lons,
+            resolution=config.resolution,
+            observation_dim=config.observation_dim,
+            analysis_dim=config.analysis_dim,
+            node_dim=config.node_dim,
+            edge_dim=config.edge_dim,
+            num_blocks=config.num_blocks,
+            hidden_dim_processor_node=config.hidden_dim_processor_node,
+            hidden_dim_processor_edge=config.hidden_dim_processor_edge,
+            hidden_layers_processor_node=config.hidden_layers_processor_node,
+            hidden_layers_processor_edge=config.hidden_layers_processor_edge,
+            hidden_dim_decoder=config.hidden_dim_decoder,
+            hidden_layers_decoder=config.hidden_layers_decoder,
+            norm_type=config.norm_type,
+            use_checkpointing=config.use_checkpointing,
+        )
```

Comment:
This should be in the dacite config. Ideally, the `GraphWeatherAssimilatorConfig`, and all the other `*Config` ones would have a `.build()` option that builds the `GraphWeatherAssimilatorConfig`. So the code would end up like this:

```python
config = dacite.from_dict(data_class=GraphWeatherAssimilatorConfig, data=config_dict)
assimilator = config.build()
```